p2p.go

```diff
@@ -506,28 +517,40 @@ func (h *Host) Connect(ctx context.Context, target peerstore.PeerInfo) error {
 }
 
 // Broadcast sends a message to the hosts who subscribe the topic
-func (h *Host) Broadcast(topic string, data []byte) error {
+func (h *Host) Broadcast(ctx context.Context, topic string, data []byte) error {
 	pub, ok := h.pubs[topic]
 	if !ok {
 		return nil
 	}
-	return pub.Publish(topic, data)
+	return pub.Publish(ctx, data)
 }
 
 // Unicast sends a message to a peer on the given address
-func (h *Host) Unicast(ctx context.Context, target peerstore.PeerInfo, topic string, data []byte) error {
-	if err := h.Connect(ctx, target); err != nil {
+func (h *Host) Unicast(ctx context.Context, target core.PeerAddrInfo, topic string, data []byte) error {
+	if h.unicastBlocklist.Blocked(target.ID, time.Now()) {
+		return errors.New("peer is in blocklist at this moment")
+	}
+
+	var err error
+	defer func() {
+		if err != nil {
+			h.unicastBlocklist.Add(target.ID, time.Now())
```

Comment:
is it possible that if the network is down, and all the neighbors will be added in to blocklist, which causes an  "man-made" network down for another 10 minutes?

Reply:
if network is down, p2p won't be receiving message, so it won't send out message as reply?